packages/odie-client/src/components/send-message-input/attachment-button.tsx

```diff
@@ -81,12 +66,20 @@ export const AttachmentButton: React.FC< {
 		]
 	);
 
-	if ( ! chat.conversationId || ! inferredClientId ) {
-		return null;
-	}
+	useEffect( () => {
+		if ( externalFile ) {
+			onFileUpload( externalFile );
+		}
+	}, [ externalFile ] );
```

Comment:
This really feels like code smell. React is very finicky with effects, and this will be called more than you want it to.